core/src/main/java/org/elasticsearch/index/reindex/Slices.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Represents the setting for the number of slices used by a BulkByScrollRequest. Valid values are positive integers and "auto". The
+ * "auto" setting defers choosing the number of slices to the request handler.
+ */
+public final class Slices implements ToXContent, Writeable {
```

Comment:
My thinking here was that rather than duplicate this logic (is `slices` a number or "auto") everywhere that distinction is relevant, I'd encapsulate it into a class. That and I prefer to avoid magic numbers if possible because they're less obvious than using a separate type.

That said, I definitely understand why this may not be desirable

>The boolean methods here could be static methods taking the int?

Do you mean something like, treat slices=0 as auto and just have this method somewhere

```java
public static boolean isAuto(int slices) {
  if (slices == 0) {
    return true
  } else if (slices > 0) {
    return false;
  } else {
    throw new InvalidArgumentException();
  }
}
```